### ISSUE TITLE
Cannot use tab key to navigate through Blueprint tabs 

### ISSUE THREAD

**Issue by faye-tan** [palantir/blueprint#6759]
#### Environment

- __Package version(s)__: 5.8.2
- __Operating System__: MacOS Sonoma 14.4
- __Browser name and version__: Edge, Chrome
- 
#### Steps to reproduce

Try to tab through the tabs in the example docs [here](https://blueprintjs.com/docs/#core/components/tabs)
 
#### Actual behavior

For some reason only the currently selected tab can be focused by tabbing through. Non-selected tabs do not get focus. 

https://github.com/palantir/blueprint/assets/35851484/9eea76cd-b885-48f4-b440-032af5b0fc88

#### Expected behavior

All tabs unless disabled should be able to be focused on by tabbing through

**Comment by invliD** [comment]
Looks like they _can_ be focused using arrow keys, but I agree this should be possible with the tab key as well.

**Comment by ericjeney** [comment]
Per https://github.com/palantir/blueprint/pull/6802, the [W3 guidance](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/) is to only allow navigating to the other tabs using arrow keys, rather than to make them tabbable, so I don't think we should do this.
